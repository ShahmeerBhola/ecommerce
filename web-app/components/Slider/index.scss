$slick-arrow-color: '#FFFFFF' !default;
$slick-prev-character: '\2190' !default;
$slick-next-character: '\2192' !default;
$slick-opacity-default: 0.75 !default;
$slick-opacity-on-hover: 1 !default;
$slick-opacity-not-active: 0.25 !default;

.slick-slider {
    box-sizing: border-box;
    display: block;
    position: relative;
    -webkit-tap-highlight-color: transparent;
    touch-action: pan-y;
    -webkit-touch-callout: none;
    user-select: none;
}

.slick-arrow {
    height: 45px;
    width: 45px;

    &.slick-prev {
        left: -75px;
    }

    &.slick-next {
        right: -75px;
    }

    &.slick-hidden {
        display: none;
    }

    &::before {
        font-size: 50px;
    }
}

.slick-prev,
.slick-next {
    background: transparent;
    border: 0;
    color: transparent;
    cursor: pointer;
    display: block;
    font-size: 0;
    height: 20px;
    line-height: 0;
    outline: none;
    padding: 0;
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    width: 20px;

    &:hover,
    &:focus {
        background: transparent;
        color: transparent;

        &::before {
            opacity: $slick-opacity-on-hover;
        }
    }

    &.slick-disabled::before {
        opacity: $slick-opacity-not-active;
    }

    &::before {
        color: $slick-arrow-color;
        // font-family: $slick-font-family;
        font-size: 20px;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        line-height: 1;
        opacity: $slick-opacity-default;
    }
}

.slick-list {
    display: block;
    margin: 0;
    outline: none;
    overflow: hidden;
    padding: 0;
    position: relative;

    &.dragging {
        cursor: pointer;
        cursor: hand;
    }
}

.slick-track {
    display: flex;
    left: 0;
    margin-left: auto;
    margin-right: auto;
    position: relative;
    top: 0;

    &::before,
    &::after {
        content: '';
        display: table;
    }

    &::after {
        clear: both;
    }

    .slick-loading & {
        visibility: hidden;
    }
}

.slick-track,
.slick-list {
    transform: translate3d(0, 0, 0);
}

.slick-slide {
    display: none;
    float: left;
    // https://stackoverflow.com/a/53131996/3902555
    height: inherit;
    min-height: 1px;

    > div {
        height: 100%;
    }

    [dir='rtl'] & {
        float: right;
    }

    img {
        display: block;
    }

    &.slick-loading img {
        display: none;
    }

    &.dragging img {
        pointer-events: none;
    }

    .slick-initialized & {
        display: block;
    }

    .slick-loading & {
        visibility: hidden;
    }

    .slick-vertical & {
        border: 1px solid transparent;
        display: block;
        height: auto;
    }
}

// alternatives variant
.sos-alternatives {
    .slick-list {
        overflow: visible !important;
    }

    .slick-arrow {
        background: rgba(0, 0, 0, 0.7);
        bottom: 0;
        height: auto;
        top: 0;
        transform: none;
        width: 2.5rem;
        z-index: 99;

        @media (max-width: 640px) {
            display: none !important;
        }

        &::before {
            border: 2px solid;
            content: '';
            display: block;
            height: 1rem;
            transform: rotate(45deg);
            width: 1rem;
        }
    }

    .slick-prev {
        left: 0.5rem;

        &::before {
            border-color: transparent transparent #fff #fff;
            margin: 0 0.75rem;
        }
    }

    .slick-next {
        left: calc(50% + 50vw);
        transform: translateX(-100%);

        &::before {
            border-color: #fff #fff transparent transparent;
            margin: 0 0.25rem;
        }
    }

    .slick-disabled {
        opacity: 0;
        transition: opacity 0.3s ease;
    }

    .slick-track {
        margin: 0;
    }
}

.sos-product-variants {
    .slick-arrow {
        bottom: 0;
        height: auto;
        top: 0;
        transform: none;
        width: auto;
        z-index: 30;
        display: flex !important;
        justify-content: center;
        align-items: center;

        &:focus {
            outline: none;
        }

        @media (max-width: 640px) {
            display: none !important;
        }

        &::before {
            content: '';
            border: 2px solid;
            display: block;
            height: 1.25rem;
            transform: rotate(45deg);
            width: 1.25rem;
        }
    }

    .slick-prev {
        left: 0.75rem;

        &::before {
            border-color: transparent transparent #aaa #aaa;
        }
    }

    .slick-next {
        right: 0.75rem;

        &::before {
            border-color: #aaa #aaa transparent transparent;
        }
    }
}

// slick-arrow slick-prev slick-disabled
