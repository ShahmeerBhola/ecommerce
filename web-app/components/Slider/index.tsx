import React, { FC } from 'react';
import SlickSlider, { Settings as Props } from 'react-slick';

import './index.scss';

const Slider: FC<Props> = ({ children, slidesToShow = 5, infinite = true, className = 'mt-10', ...rest }) => {
    return (
        <SlickSlider
            className={className}
            dots={false}
            arrows={false}
            slidesToShow={slidesToShow}
            infinite={infinite}
            {...rest}
        >
            {children}
        </SlickSlider>
    );
};

export default Slider;
