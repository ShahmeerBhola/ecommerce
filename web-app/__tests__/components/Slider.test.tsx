import React from 'react';
import renderer from 'react-test-renderer';
import Slider from '../../components/Slider';

// was getting this error when using 0.26.1, bumping down to 0.25.2 fixed it
// cannot read property 'querySelectorAll'
// https://github.com/akiran/react-slick/issues/1830
describe('Slider component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <Slider>
          <div>
            <p>Slide 1</p>
          </div>
          <div>
            <p>Slide 2</p>
          </div>
          <div>
            <p>Slide 3</p>
          </div>
        </Slider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when infinite is true', () => {
    const tree = renderer
      .create(
        <Slider infinite>
          <div>
            <p>Slide 1</p>
          </div>
          <div>
            <p>Slide 2</p>
          </div>
          <div>
            <p>Slide 3</p>
          </div>
        </Slider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when infinite is false', () => {
    const tree = renderer
      .create(
        <Slider infinite={false}>
          <div>
            <p>Slide 1</p>
          </div>
          <div>
            <p>Slide 2</p>
          </div>
          <div>
            <p>Slide 3</p>
          </div>
        </Slider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when slidesToShow is non-default', () => {
    const tree = renderer
      .create(
        <Slider slidesToShow={1}>
          <div>
            <p>Slide 1</p>
          </div>
          <div>
            <p>Slide 2</p>
          </div>
          <div>
            <p>Slide 3</p>
          </div>
        </Slider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
