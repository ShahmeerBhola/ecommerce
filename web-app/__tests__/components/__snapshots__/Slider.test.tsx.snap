// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component it renders properly 1`] = `
<div
  className="slick-slider my-20 slick-initialized"
>
  
  <div
    className="slick-list"
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-cloned"
        data-index={3}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-cloned"
        data-index={4}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={5}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  
  
</div>
`;

exports[`Slider component it renders properly when infinite is false 1`] = `
<div
  className="slick-slider my-20 slick-initialized"
>
  
  <div
    className="slick-list"
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  
  
</div>
`;

exports[`Slider component it renders properly when infinite is true 1`] = `
<div
  className="slick-slider my-20 slick-initialized"
>
  
  <div
    className="slick-list"
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-cloned"
        data-index={3}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-cloned"
        data-index={4}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={5}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  
  
</div>
`;

exports[`Slider component it renders properly when slidesToShow is non-default 1`] = `
<div
  className="slick-slider my-20 slick-initialized"
  dir="ltr"
>
  <div
    className="slick-arrow slick-prev"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <svg
      fill="none"
      height="72"
      viewBox="0 0 38 72"
      width="38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38 2L4 36L38 70L36 72L0 36L36 0L38 2Z"
        fill="#C5C5C5"
      />
    </svg>
  </div>
  <div
    className="slick-list"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={null}
    onMouseMove={null}
    onMouseUp={[Function]}
    onTouchCancel={null}
    onTouchEnd={[Function]}
    onTouchMove={null}
    onTouchStart={[Function]}
    style={Object {}}
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={-1}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={3}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 1
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={4}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 2
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={5}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <p>
              Slide 3
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="slick-arrow slick-next"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
  >
    <svg
      fill="none"
      height="72"
      viewBox="0 0 38 72"
      width="38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 2L34 36L0 70L2 72L38 36L2 0L0 2Z"
        fill="#C5C5C5"
      />
    </svg>
  </div>
</div>
`;
